fasteners/threads.scad

```diff
@@ -44,32 +59,140 @@ use <scad-utils/transformations.scad>
 
 
 // ----------------------------------------------------------------------------
-// internal - true = clearances for internal thread (e.g., a nut).
-//            false = clearances for external thread (e.g., a bolt).
-//            (Internal threads should be "cut out" from a solid using
-//            difference()).
-// n_starts - Number of thread starts (e.g., DNA, a "double helix," has
-//            n_starts=2).  See wikipedia Screw_thread.
+// Metric Threads in accordance with ISO 68-1
+// ----------------------------------------------------------------------------
+// diameter  - nominal diameter of thread in mm. (e.g. 10 for an M10 thread)
+// pitch     - distance in mm from the crest of one thread to the next.
+// length    - overall axial length of thread in mm.
+// internal  - true = clearances for internal thread (e.g., a nut).
+//             false = clearances for external thread (e.g., a bolt).
+//             (Internal threads should be "cut out" from a solid using
+//             difference()).
+// clearance - additional clearance to add in mm. Teeth are effectively
+//             shifted inwards (outwards) for external (internal) threads by
+//             this amount.
+// n_starts  - Number of thread starts (e.g., DNA, a "double helix," has
+//             n_starts=2).  See Wikipedia Screw_thread.
 module metric_thread (
     diameter = 8,
     pitch = 1,
     length = 1,
     internal = false,
+    clearance = 0,
     n_starts = 1
```

Comment:
I know we probably don't want to break current implementations but, to me it feels like we should drop the `n_starts` parameter here, ISO 68-1 threads always are single-start threads.

Reply:
While that may be the case, I think this module already allows us to generate off-spec threads (pitch-diameter combinations that aren't in any standard), so I'm inclined to keep `n_starts` for other (ab)uses. I've personally been (ab)using `metric_thread` for threaded containers of arbitrary dimensions. I can see the use in having multi-start threads for those, and using `trapezoidal_thread` requires extra calculations to achieve the same effect.